research-app/src/TransitionExpand.vue

```diff
@@ -0,0 +1,93 @@
+<template>
+  <!-- This component is based on the code from https://markus.oberlehner.net/blog/transition-to-height-auto-with-vue/
+  It has been modified to use the vue-property-decorator format
+  -->
+  <transition
+    name="expand"
+    @enter="enter"
+    @after-enter="afterEnter"
+    @leave="leave"
+  >
+    <slot/>
+  </transition>
+</template>
+
+<script lang="ts">
+import { Component,  Emit,  Vue } from 'vue-property-decorator';
+
+@Component
+export default class TransitionExpand extends Vue {
+  name = 'TransitionExpand';
+
+  @Emit()
+  enter(element: HTMLElement) {
+    const width = getComputedStyle(element).width;
+
+    element.style.width = width;
+    element.style.position = 'absolute';
+    element.style.visibility = 'hidden';
+    element.style.height = 'auto';
+
+    const height = getComputedStyle(element).height;
+
+    element.style.width = '';
+    element.style.position = '';
+    element.style.visibility = 'visible';
+    element.style.height = '0px';
+
+    // Force repaint to make sure the
+    // animation is triggered correctly.
+    getComputedStyle(element).height;
+
+    // Trigger the animation.
+    // We use `requestAnimationFrame` because we need
+    // to make sure the browser has finished
+    // painting after setting the `height`
+    // to `0` in the line above.
+    requestAnimationFrame(() => {
+    element.style.height = height;
+    });
+  }
+
+  @Emit()
+  afterEnter(element: HTMLElement) {
+    element.style.height = 'auto';
+  }
+
+  @Emit()
+  leave(element: HTMLElement) {
+    const height = getComputedStyle(element).height;
+    
+    element.style.height = height;
+
+    // Force repaint to make sure the
+    // animation is triggered correctly.
+    getComputedStyle(element).height;
+
+    requestAnimationFrame(() => {
+      element.style.height = '0';
+    });
+  }
+
+}
+</script>
+
+<style scoped>
+* {
+  will-change: height;
+  transform: translateZ(0);
+  backface-visibility: hidden;
+  perspective: 1000px;
+}
+
+.expand-enter-active,
+.expand-leave-active {
+  transition: height 0.2s ease-in-out;
+  overflow: hidden;
+}
+
+.expand-enter,
+.expand-leave-to {
+  height: 0;
+}
+</style>
```

Comment:
This little icon here means that this file does not end with a newline character. "Best practice" (in Unix-land at least) is to include one; most edtiors have an option to enforce that there always is one.

Reply:
Yep, there's a setting for that in VS Code - shouldn't be a problem in the future.